### PR TITLE
choromanski2020 rethinking attention

### PR DESCRIPTION
**Pull request by joshpapermaster**

### CODE REVIEW

File: _summaries/choromanski2020rethinking_1.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+The authors suggest a new algorithm for performers called FAVOR+, which stands for fast attention via positive orthogonal random features. 
+
+Using this approach brings the time complexity of attention from O(L^2 * D) to O(Lrd)
+- This is beneficial because we can control r as a parameter and decrease the computational effort needed.
+
+![favor+](choromanski2020rethinking_1b.png)
+
+FAVOR+ uses a feature map to represent the kernel to keep time complexity low.
+
+The similarities between the guassian kernel and the softmax kernel are noted, but guassian kernels can include negative values which leads to unstable results.
```
Comment (DartingMelody) [inline _summaries/choromanski2020rethinking_1.md:27]: Can you add maybe - Therefore, kernel producing non-negative scores are used in this model.  
Also, can you add how they are similar or different a bit more?

File: _summaries/choromanski2020rethinking_1.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- This is beneficial because we can control r as a parameter and decrease the computational effort needed.
+
+![favor+](choromanski2020rethinking_1b.png)
+
+FAVOR+ uses a feature map to represent the kernel to keep time complexity low.
+
+The similarities between the guassian kernel and the softmax kernel are noted, but guassian kernels can include negative values which leads to unstable results. 
+
+The feature map allows for lower time complexity and the paper cites a recent publication of the gaussian kernel being represented by using a specific feature map. So this is expanded to include only positive results and stabilize the kernel. 
+
```
Comment (DartingMelody) [inline _summaries/choromanski2020rethinking_1.md:30]: Maybe you can add about the variance of the new unbiased positive random feature map estimators tend to be 0 as approximated values tend to zero. Thus, proposing a robust mechanism approximating attention.

File: _summaries/choromanski2020rethinking_1.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+It was also proven that FAVOR+ produces better results than the IID and trigonometric features:
+![favor+](choromanski2020rethinking_1f.png)
+
+Experiments proved that performers worked with already trained models with minor finetuning. 
+
+Additionally, the Performer RELU had significantly higher accuracy than both Reformer and Linformer on the bidirectional and undirectional cases of the protein sequences from the TrEMBL. The Performer softmax also had good results, and received the same results as the original softmax.
+
+## TL;DR
+- Performers provide a time and space efficient approximation of attention using the FAVOR+ approach
```
Comment (DartingMelody) [inline _summaries/choromanski2020rethinking_1.md:60]: you may add - with low estimation variance, strong theoretical guarantees and uniform convergence

**Review by DartingMelody — COMMENTED** [review]
Hi, 
This is a good summary, quite easy to flow. I have added some comments, let me know if you have any doubts or need any help.

**Comment by joshpapermaster** [comment]
> Hi, This is a good summary, quite easy to flow. I have added some comments, let me know if you have any doubts or need any help.

Hey, just uploaded my revisions. Thanks for the review

**Comment by DartingMelody** [comment]
> Hey, just uploaded my revisions. Thanks for the review

LGTM

**Review by zhaoyue-zephyrus — APPROVED** [review]

**Pull request merged**